scylla/src/statement/prepared_statement.rs

```diff
@@ -527,22 +533,22 @@ mod tests {
                 ColumnType::BigInt,
                 ColumnType::Blob,
             ],
-            [4, 0, 3],
+            [0, 3, 4],
```

Comment:
Sorry, I got confused previously. We should explicitly sort the `PartitionKeyIndex` structures by `index` after all. The order `4, 0, 3` was intentional here - you can have a query which expects partition key column to be specified out of order (e.g. `UPDATE tbl SET v = ? WHERE pk3 = ? AND pk1 = ? AND pk2 = ?`).

Reply:
I don't think I get your point.
How should I adjust the code then? Before it was ..., now it has to be ... ?

Reply:
From `frame/response/result.rs`:

```rust
pub struct PreparedMetadata {
    // ...
    /// pk_indexes are sorted by `index` and can be reordered in partition key order
    /// using `sequence` field
    pub pk_indexes: Vec<PartitionKeyIndex>,
    // ...
}
```

Reply:
I knew the snippet, yet I'm still confused.

Reply:
```rust
let meta = make_meta(
    [
        ColumnType::TinyInt,
        ColumnType::SmallInt,
        ColumnType::Int,
        ColumnType::BigInt,
        ColumnType::Blob,
    ],
    [4, 0, 3],
);
```

The above metadata was supposed to simulate the case where you have a query that constructs the partition key from 4th, 0th and 3rd bind marker - e.g.:

```cql
INSERT INTO tbl (pk2, v1, v2, pk3, pk1) VALUES (?, ?, ?, ?, ?);
```

So, the order of the numbers in the second argument was intentional, and requiring the numbers to be sorted is wrong. The `make_meta` is incorrect and should sort the generated pk indexes itself, as it is suggested by the comment in `PreparedMetadata` and as it is actually done in the deserialization code.

The test currently asserts the wrong thing. As a result, there is a bug in `PartitionKeyDecoder` which manifests when you have multiple partition key columns with different types and they specified out of order in the bind markers. I'll send a fix, then let's rebase.

Reply:
Rebased.